src/main/java/org/elasticsearch/index/query/QueryBuilder.java

```diff
@@ -45,4 +45,10 @@
      * @throws IOException
      */
     Query toQuery(QueryParseContext parseContext) throws QueryParsingException, IOException;
+
+    /**
+     * Validate the query.
+     * @throws ElasticsearchException if the query is not valid, e.g. fields that are needed to create the lucene query are missing.
+     */
+    void validate() throws ElasticsearchException;
```

Comment:
indeed I suggest to move away from `Preconditions`, for now using ActionRequestValidationException sounds good, maybe a specific query exception would be better but we can figure it out later on I guess